hbase-common/src/test/java/org/apache/hadoop/hbase/HBaseClassTestRuleChecker.java

```diff
@@ -41,9 +41,12 @@
   @Override
   public void testStarted(Description description) throws Exception {
     Category[] categories = description.getTestClass().getAnnotationsByType(Category.class);
-    for (Class<?> c : categories[0].value()) {
-      if (c == IntegrationTests.class) {
-        return;
+    // Don't fail if there is a missing category
+    if (categories.length > 0) {
+      for (Class<?> c : categories[0].value()) {
```

Comment:
Shouldn't we check all the categories instead of the first?

Reply:
I'm not actually sure what all is allowed by `Category`. I would assume that this for-loop handles both `@Category(OneInterface.class)` and `@Category({InterfaceA.class,InterfaceB.class})`. I have to experiment to see what else is actually allowable.

Reply:
The `@Category` annotation is not `Repeatable` so we can only have a single one. Its value is `Class<?>[]`, so we can't provide anything else, e.g. an array of class<?>[] is disallowed. I think this is OK, but will add an explicit check on this :)